lib/teams/index.ts

```diff
@@ -294,3 +298,185 @@ export class PlatformTeam extends ApplicationTeam {
+     */
+  constructor(props: EmrEksTeamProps) {
+    super(props);
+    this.emrTeam = props;
+  }
+
+  setup(clusterInfo: ClusterInfo): void {
+    const cluster = clusterInfo.cluster;
+
+    const emrVcPrerequisit = this.setEmrContainersForNamespace(cluster, this.emrTeam.virtualClusterNamespace, this.emrTeam.createNamespace);
+
+    this.emrTeam.excutionRoles.forEach(excutionRole => {
+
+      const executionRolePolicy = excutionRole.excutionRoleIamPolicy ?
+        excutionRole.excutionRoleIamPolicy :
+        new ManagedPolicy(cluster.stack, `executionRole-${excutionRole.excutionRoleName}-Policy`, {
+          statements: excutionRole.excutionRoleIamPolicyStatement,
+        });
+
+      this.createExecutionRole(
+        cluster,
+        executionRolePolicy,
+        this.emrTeam.virtualClusterNamespace,
+        excutionRole.excutionRoleName);
+    });
+
+    const teamVC = new CfnVirtualCluster(cluster.stack, `${this.emrTeam.virtualClusterName}-VirtualCluster`, {
+      name: this.emrTeam.virtualClusterName,
+      containerProvider: {
+        id: cluster.clusterName,
+        type: 'EKS',
+        info: { eksInfo: { namespace: this.emrTeam.virtualClusterNamespace } },
+      },
+      tags: [{
+        key: 'created-with',
+        value: 'cdk-blueprint',
+      }],
+    });
+
+    teamVC.node.addDependency(emrVcPrerequisit);
+
+    new CfnOutput(cluster.stack, `${this.emrTeam.virtualClusterName}-id`, {
+      value: teamVC.attrId
+    });
+
+  }
+  /**
+   * @internal
+   * Private method to to apply k8s RBAC to the service account used by EMR on EKS service role
+   * For more information on the RBAC read consult the EMR on EKS documentation in this link 
+   * https://docs.aws.amazon.com/emr/latest/EMR-on-EKS-DevelopmentGuide/setting-up-cluster-access.html
+   * This method 
+   * @param cluster EKS cluster where to apply the RBAC
+   * @param namespace Namespace where the RBAC are applied
+   * @param createNamespace flag to create namespace if not already created
+   * @returns 
+   */
+
+  private setEmrContainersForNamespace(cluster: Cluster, namespace: string, createNamespace: boolean): KubernetesManifest {
+
+    //Get the Role definition and add the namespace of the EMR on EKS virtual cluster
+    let emrContainersK8sRole = readYamlDocument(`${__dirname}/emrContainersRole.yaml`);
+    emrContainersK8sRole = emrContainersK8sRole.replace('<REPLACE-NAMESPACE>', namespace);
+    const emrContainersK8sRoleManifest = loadYaml(emrContainersK8sRole);
+
+    //Create the role used by EMR on EKS
+    const emrContainersK8sRoleResource = cluster.addManifest('emrContainersK8sRoleManifest',
+      emrContainersK8sRoleManifest
+    );
+
+    //Get the RoleBinding definition and add the namespace of the EMR on EKS virtual cluster  
+    let emrContainersK8sRoleBinding = readYamlDocument(`${__dirname}/emrContainersRoleBinding.yaml`);
+    emrContainersK8sRoleBinding = emrContainersK8sRoleBinding.replace('<REPLACE-NAMESPACE>', namespace);
```

Comment:
For generic manifests, please use `KubectlProvider. applyManifestDeployment` - it allows passing replaceable parameters as {{key}}. Example [here](https://github.com/aws-quickstart/cdk-eks-blueprints/blob/main/lib/addons/amp/index.ts#L114). this is needed not just for consistency.

Reply:
applies elsewhere.

Reply:
Fixed, but could not use the method you shared. The example you shared is for Helm Add On. I used the method described here in the [doc](https://aws-quickstart.github.io/cdk-eks-blueprints/extensibility/#non-helm-add-ons)

Reply:
@lmouhib the example I shared for non-helm add-on, you can see that that class just `implements ClusterAddOn`. The doc that you referred to was for external add-on providers, however I think we should update it as well.

Reply:
Done, I was calling `KubectlProvider` from the CDK, not from the AddOn library and it was having a weird behavior.
This is fixed and tested.